language/move-compiler/src/typing/core.rs

```diff
@@ -1703,3 +1728,23 @@ fn check_num_tvar_(subst: &Subst, loc: Loc, ty: &Type) -> bool {
         _ => false,
     }
 }
+//**************************************************************************************************
+// Function type check
+//**************************************************************************************************
+
+pub fn check_non_fun(context: &mut Context, ty: &Type) {
+    if let sp!(
+        loc,
+        Type_::Apply(
+            _,
+            sp!(_, TypeName_::Builtin(sp!(_, BuiltinTypeName_::Fun))),
+            _
+        )
+    ) = ty
```

Comment:
You might need to unfold here, but I would need to see the usages

Reply:
What means unfolding in this context?